test/rekt/features/channel/control_plane.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package channel
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
+	messagingclientsetv1 "knative.dev/eventing/pkg/client/clientset/versioned/typed/messaging/v1"
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/state"
+)
+
+func ControlPlaneConformance() *feature.FeatureSet {
+	fs := &feature.FeatureSet{
+		Name: "Knative Channel Specification - Control Plane",
+		Features: []feature.Feature{
+			*ControlPlaneChannel(),
+		},
+	}
+
+	return fs
+}
+
+func todo(ctx context.Context, t feature.T) {
+	t.Log("TODO, Implement this.")
+}
+
+func ControlPlaneChannel() *feature.Feature {
+	f := feature.NewFeatureNamed("Conformance")
+
+	f.Stable("Aggregated Channelable Manipulator ClusterRole").
+		Must("Every CRD MUST create a corresponding ClusterRole, that will be aggregated into the channelable-manipulator ClusterRole.", todo).
+		Must("This ClusterRole MUST include permissions to create, get, list, watch, patch, and update the CRD's custom objects and their status.", todo).
+		Must("Each channel MUST have the duck.knative.dev/channelable: \"true\" label on its channelable-manipulator ClusterRole.", todo)
+
+	f.Stable("Aggregated Addressable Resolver ClusterRole").
+		Must("Every CRD MUST create a corresponding ClusterRole, that will be aggregated into the addressable-resolver ClusterRole.", todo).
+		Must("This ClusterRole MUST include permissions to get, list, and watch the CRD's custom objects and their status.", todo).
+		Must("Each channel MUST have the duck.knative.dev/addressable: \"true\" label on its addressable-resolver ClusterRole.", todo)
+
+	f.Stable("CustomResourceDefinition per Channel").
+		// Each channel is namespaced and MUST have the following: TODO: seems like we wanted to say channels must be namespaced?
+		Must("label of messaging.knative.dev/subscribable: true", todo).
+		Must("label of duck.knative.dev/addressable: true", todo).
+		Must("The category `channel`", todo)
+
+	f.Stable("Annotation Requirements").
+		Should("have annotation: messaging.knative.dev/subscribable: [v1beta1 || v1]", todo)
+
+	f.Stable("Spec Requirements - v1beta1").
```

Comment:
just v1? since we're doing conformance, we should target v1? Here and elsewhere?

Reply:
it is in the spec? idk

Reply:
Pending on https://github.com/knative/specs/pull/12

Reply:
We should remove it, since we're removing it in the release after next.